branding/src/main/scala/com/normation/plugins/branding/BrandingConfService.scala

```diff
@@ -73,57 +77,66 @@ class BrandingConfService extends Loggable {
   val configFilePath = "/var/rudder/plugins/branding/configuration.json"
 
 
-  private[this] var cache : Box[BrandingConf] = reloadCache(true)
+  private[this] lazy val cache : Ref[Either[RudderError, BrandingConf]] = (for {
+    v <- Ref.make[Either[RudderError, BrandingConf]](Left(Unexpected("Cache is not yet initialized")))
+    c <- reloadCacheInternal(true, v).either
+  } yield v).runNow
+
+  private def reloadCacheInternal(init: Boolean, ref: Ref[Either[RudderError, BrandingConf]]) : IOResult[BrandingConf] = {
+    (for {
+      path   <- IOResult.effect(Paths.get(configFilePath))
+      exists <- IOResult.effect(Files.exists(path))
+      res    <- if(exists) {
+                  for {
+                    content <- IOResult.effect(s"Error when trying to read file: ${configFilePath}")(
+                                 File(configFilePath).contentAsString(StandardCharsets.UTF_8)
+                               )
+                    json    <- parseOpt(content).notOptional("Could nor parse correctly Branding plugin configuration file")
+                    conf    <- BrandingConf.parse(json).toIO
+                    ref     <- ref.set(Right(conf))
+                  }  yield {
+                    conf
+                  }
+                } else {
+                  if (init) {
+                    updateConf(initialValue)
+                  } else {
+                    // Should we update cache to that value ??
```

Comment:
If it happened after boot, it's likely something the user want to be notifier loudly about. At worse, he can restart rudder (and it will go back to init).